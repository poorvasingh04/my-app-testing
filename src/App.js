import logo from './logo.svg';
import './App.css';
import { initializeApp } from "firebase/app";
import { useEffect } from 'react';

// Initialize Firebase
const firebaseConfig = {
  apiKey: "AIzaSyC9WcbEUlG-f2wmmMNoeaSfSzZFJN3DdIM",
  authDomain: "my-app-e6ecd.firebaseapp.com",
  projectId: "my-app-e6ecd",
  storageBucket: "my-app-e6ecd.appspot.com",
  messagingSenderId: "254101470394",
  appId: "1:254101470394:web:e58ff669b5f998a00c9683"
};
initializeApp(firebaseConfig);

function App() {
  useEffect(() => {

  }, []);

  return (
    <div className="App">
      <header className="App-header">
        <img src={logo} className="App-logo" alt="logo" />
        <p>
          This is my project my-app. Hello
        </p>
        <a
          className="App-link"
          href="https://reactjs.org"
          target="_blank"
          rel="noopener noreferrer"
        >
          Learn React
        </a>
      </header>
    </div>
  );
}

export default App;
