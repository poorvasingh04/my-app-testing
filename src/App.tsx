import React, { ReactElement } from "react";
import { initializeApp } from "firebase/app";
import { useEffect } from "react";
import { BrowserRouter as Router, Route, Routes } from "react-router-dom";
import BootstrapComponent from "./components/bootstrap/Components";
import TailwindComponent from "./components/tailwind/Components";

import Home from "./components/Home";
import "./App.css";

function App(): ReactElement<React.FC> {
  useEffect(() => {
    // Initialize Firebase
    const firebaseConfig = {
      apiKey: process.env.REACT_APP_FIREBASE_APIKEY,
      authDomain: process.env.REACT_APP_FIREBASE_AUTHDOMAIN,
      projectId: process.env.REACT_APP_FIREBASE_PROJECTID,
      storageBucket: process.env.REACT_APP_FIREBASE_STORAGE_BUCKET,
      messagingSenderId: process.env.REACT_APP_FIREBASE_MESSAGING_SENDERID,
      appId: process.env.REACT_APP_FIREBASE_APPID,
    };

    initializeApp(firebaseConfig);
  }, []);

  return (
    <div className="App">
      <Router>
        <Routes>
          <Route path="/" Component={Home} />
          <Route path="/bootstrap" Component={BootstrapComponent} />
          <Route path="/tailwind" Component={TailwindComponent} />
        </Routes>
      </Router>
    </div>
  );
}

export default App;
