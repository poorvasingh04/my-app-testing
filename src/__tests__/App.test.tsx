import React from "react";
import renderer from "react-test-renderer";
import "@testing-library/jest-dom";
import App from "../App";

it("test 1", () => {
  process.env.PUBLIC_URL = "dev";
  const component = renderer.create(<App />);

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
