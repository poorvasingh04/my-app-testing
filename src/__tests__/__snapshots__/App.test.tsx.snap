// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test 1 1`] = `
<div
  className="App"
>
  <div>
    <a
      href="/component"
      onClick={[Function]}
    >
      <p>
        Bootstrap
      </p>
    </a>
    <a
      href="/tailwind"
      onClick={[Function]}
    >
      <p>
        Tailwind
      </p>
    </a>
  </div>
</div>
`;
